src/components/AudioPlayer/QuranReaderHighlightDispatcher.tsx

```diff
@@ -104,15 +107,18 @@ const useMemoizedHighlightedWordLocation = (
   const lastHighlightedWordLocation = useRef<Segment>(null);
   if (!currentHighlightedVerseTiming) return null;
 
+  // do not highlight verse when currentTime is 0
```

Comment:
// Do not highlight the verse when the audio hasn't started playing
if (currentTime === 0) return defaultState/initialState;

Reply:
Also, let's do this check in the top part of the file.

This way we can short circuit the unnecessary code.

Reply:
👍  updated the docs to be more clear.

and this function needs to return `null | VerseTiming`. The final result will be `defaultState` when this function returns null 👍